src/Altinn.Profile.Core/Utils/OptionalJsonConverter.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Altinn.Profile.Core.Utils
+{
+    /// <summary>
+    /// Converts <see cref="Optional{T}"/> values to and from JSON, handling explicit nulls and missing values.
+    /// </summary>
+    /// <typeparam name="T">The type of the value contained in the <see cref="Optional{T}"/>.</typeparam>
+    public class OptionalJsonConverter<T> : JsonConverter<Optional<T>>
```

Comment:
Could we try to create a few unit tests on this or does the controller tests cover everything?